contracts/src/interfaces/ITeleporterTokenSource.sol

```diff
@@ -0,0 +1,55 @@
+// (c) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+// SPDX-License-Identifier: Ecosystem
+
+pragma solidity 0.8.18;
+
+import {
+    ITeleporterTokenBridge, SendTokensInput, SendAndCallInput
+} from "./ITeleporterTokenBridge.sol";
+
+/**
+ * THIS IS AN EXAMPLE CONTRACT THAT USES UN-AUDITED CODE.
+ * DO NOT USE THIS CODE IN PRODUCTION.
+ */
+
+/**
+ * @dev Interface for a "home" or "source" bridge contract that locks
+ * tokens on its chain to be bridge to supported destination bridge contracts on other chains.
+ */
+interface ITeleporterTokenSource is ITeleporterTokenBridge {
+    /**
+     * @dev Emitted when tokens are added as collateral for a destination bridge.
```

Comment:
Can you add a comment that `CollateralAdded` is guaranteed to emit an event with `remaining == 0` on the call to `addCollateral` that fully collateralizes the bridge? I think that particular event will be what services that integrate with the bridge listen for to decide what function to call. 

We could also add a standalone `BridgeCollateralized` event as well.

Reply:
added comment about when remaining is zero meaning collateralized

Reply:
Just calling out that if the initial reserve imbalance is 0 (which is the case for all ERC20Destinations), then no `CollateralAdded` events will be emitted for that destination all. I think the comment you added is good though 👍

Reply:
good call out